src/runtime/GlobalObjectBuiltinTypedArray.cpp

```diff
@@ -932,7 +940,9 @@ static Value builtinTypedArraySubArray(ExecutionState& state, Value thisValue, s
     // Let argumentsList be «buffer, beginByteOffset, newLength».
     Value args[3] = { buffer, Value(beginByteOffset), Value(newLength) };
     // Return Construct(constructor, argumentsList).
-    return Object::construct(state, constructor, 3, args);
+    Value A = Object::construct(state, constructor, 3, args);
+    validateTypedArray(state, A, state.context()->staticStrings().subarray.string());
```

Comment:
How about adding a new method for `TypedArraySpeciesCreate` [22.2.4.7](https://www.ecma-international.org/ecma-262/10.0/#typedarray-species-create) since this method is invoked in many places like %TypedArray%.prototype.filter, %TypedArray%.prototype.map etc.